### PR TITLE
Remove NET452 NET46 from OpenTelemetry sdk

### PR DESCRIPTION
**Pull request by cijothomas**
Part of https://github.com/open-telemetry/opentelemetry-dotnet/issues/2138

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by utpilla — APPROVED** [review]

**Comment by cijothomas** [comment]
Waiting for one more approval, as this is bit larger PR, and it touches the heart of this repo!

File: src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs [hunk 1]
```diff
@@ -324,11 +324,7 @@ protected override void OnEventWritten(EventWrittenEventArgs e)
                 {
                     message = e.Message;
                 }
-#if NET452
-                Debug.WriteLine($"{e.EventSource.Name} - EventId: [{e.EventId}], Message: [{message}]");
-#else
-                Debug.WriteLine($"{e.EventSource.Name} - EventId: [{e.EventId}], EventName: [{e.EventName}], Message: [{message}]");
-#endif
+Debug.WriteLine($"{e.EventSource.Name} - EventId: [{e.EventId}], EventName: [{e.EventName}], Message: [{message}]");
```
Comment (reyang) [inline src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs:327]: Indent?

File: src/OpenTelemetry/Internal/SelfDiagnosticsEventListener.cs [hunk 2]
```diff
@@ -55,11 +55,7 @@ public SelfDiagnosticsEventListener(EventLevel logLevel, SelfDiagnosticsConfigRe
 
             foreach (var eventSource in eventSources)
             {
-#if NET452
-                this.EnableEvents(eventSource, this.logLevel, (EventKeywords)(-1));
```
Comment (reyang) [inline src/OpenTelemetry/Internal/SelfDiagnosticsEventListener.cs:59]: Finally 👍 

IIRC there are many places where we have `(EventKeywords)(-1)`, probably not fully covered by this PR. Can be a separate PR though.

File: src/OpenTelemetry/OpenTelemetry.csproj [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net452;net46;net461</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
```
Comment (reyang) [inline src/OpenTelemetry/OpenTelemetry.csproj:3]: After this PR we might be able to consolidate these `TargetFrameworks` (e.g. using the default setting in the top level `.props` file).

**Review by reyang — APPROVED** [review]
LGTM.

**Pull request merged**